python/ray/tune/trainable.py

```diff
@@ -632,6 +634,16 @@ def get_config(self):
         """Returns configuration passed in by Tune."""
         return self.config
 
+    def build_config(self, config: Dict):
+        """Builds a deep copy of the input config and populates it with
+        metadata from this Trainable.
+        """
+        new_config = copy.deepcopy(config)
```

Comment:
I don't think this logic should belong in the `Trainable` base class. Populating the configs with the appropriate attributes is unique to the DistributedTrainables because they are `Trainables` that wrap `FunctionTrainables` so they have to propagate this information from the former to the latter. But this isn't necessary for all `Trainables` in general. I would say that moving this logic to another `DistributedTrainable` class that the TF,Torch,Horovod ones inherit from, or just copying the 4 lines for all 3 trainables would be better options here.

Reply:
Thanks for the suggestion, that makes a lot of sense. I created a new `DistributedTrainable` class to hold this method, can you take another look to see if the new inheritance model is organized correctly?